base/setup/lib/CMakeLists.txt

```diff
@@ -33,3 +33,9 @@ list(APPEND SOURCE
 add_library(setuplib ${SOURCE})
 add_pch(setuplib precomp.h SOURCE)
 add_dependencies(setuplib xdk) # psdk
+
+
+## Unit-tests
+if(ENABLE_ROSTESTS)
+    add_subdirectory(unittests)
+endif()
```

Comment:
It would probably be better to move all the tests into the `rostests` directory.

Reply:
Indeed but I have the impression this doesn't ensure component compartmentalization at the source level, hence why I experimented this way.